### DISCUSSION TITLE
Not able to verify digital signature

### DISCUSSION THREAD

**Discussion by sourjya123** [micronaut-projects/micronaut-core#9800] [General]
Hello,
I am using micronaut framework with kotlin code. I am trying to verify the digital signature where the signature is present in the header with the request body. The verification is written in doFilter() method. But while I am doing request.body instead of giving me the same request body sent by postman or any client, micronaut framework is converting it to a class object based on the body argument. Is there any way to verify the signature before micronaut does the mapping to a class obj.

**Comment by yawkat** [comment]
if you are using micronaut 4, you can use the new request filter method API, and declare your filter like this:

```java
    @RequestFilter
    public void filterRequest(HttpRequest<?> request, @Body byte[] body) {
```
